deps/rabbitmq_prometheus/docker/docker-compose-overview.yml

```diff
@@ -173,6 +170,11 @@ services:
       CONSUMERS: 0
       SERVERS_STARTUP_TIMEOUT: *startup_timeout
       METRICS_PROMETHEUS: "true"
+  stream:
+    image: pivotalrabbitmq/stream-perf-test
+    networks:
+      - "rabbitmq-prometheus"
+    command: "--uris rabbitmq-stream://guest:guest@rmq2:5552/%2f --max-length-bytes 100mb --rate 1000"
```

Comment:
When I run `make overview`, container `docker_stream_1` gets created but exits immediately with
```
> docker logs docker_stream_1
...
Connection refused: rmq2/172.21.0.12:5552
...
```
Therefore, I'm unable to see any metrics in the Grafana Streams dahsboard.

(Running `make overview` a second time after a couple of seconds will succeed and will show up stream metrics.)

Adding
```
    depends_on:
      - "rmq2"
```
does not help since `rmq2` takes too long to open port 5552.

I suggest to either make stream-perf-test retry to connect or to add a wait wrapper script to the stream-perf-test image as described in https://docs.docker.com/compose/startup-order/ (cc @acogoluegnes).

Reply:
We can also merge this PR. We should make this change in https://github.com/rabbitmq/rabbitmq-stream-java-client/blob/main/Dockerfile ?

Reply:
Good spot, let me know if https://github.com/rabbitmq/rabbitmq-server/pull/3250/commits/35a6369327a07b1957550b2fb6e57ab34053a054 fixes it for you too.

Reply:
I just tried adding a one-liner `        wait-for-it \` to https://github.com/rabbitmq/rabbitmq-stream-java-client/blob/8866cebafda34c15104f3557b3df8d74037abcfb/Dockerfile#L67 and changed the docker compose file to:
```
  stream:
    image: pivotalrabbitmq/stream-perf-test
    networks:
      - "rabbitmq-prometheus"
    depends_on:
      - "rmq2"
    entrypoint: ["wait-for-it"]
    command: ["rmq2:5552", "--timeout=180", "--",
        "java", "-Dio.netty.processId=1", "-jar", "stream-perf-test.jar",
        "--uris", "rabbitmq-stream://guest:guest@rmq2:5552/%2f",
        "--max-length-bytes", "100mb",
        "--rate", "1000"]

```
which works fine. It takes around 20 seconds until the port is open.

Your approach in https://github.com/rabbitmq/rabbitmq-server/commit/35a6369327a07b1957550b2fb6e57ab34053a054 works for me. It restarts the container a couple of times which is fine. It's the far simper option :)